crates/walrus-sdk/src/client/quilt_client.rs

```diff
@@ -189,14 +215,14 @@ impl<T: ReadClient> QuiltClient<'_, T> {
         &self,
         metadata: &VerifiedBlobMetadataWithId,
         certified_epoch: Epoch,
-        first_sliver: &[SliverData<V::SliverAxis>],
+        first_sliver: &SliverData<V::SliverAxis>,
     ) -> ClientResult<QuiltIndex>
     where
         SliverData<V::SliverAxis>: TryFrom<Sliver>,
     {
         let mut all_slivers = Vec::new();
-        let mut refs = Vec::new();
-        let first_sliver_refs: Vec<_> = first_sliver.iter().collect();
+        let mut refs = [first_sliver].to_vec();
+        let first_sliver_refs = [first_sliver].to_vec();
```

Comment:
Question: (Why) do we need these three separate vectors?

Also, nit: You can simplify `[x].to_vec()` to `vec![x]`.